configman/config_manager.py

```diff
@@ -556,40 +556,45 @@ def get_option_names(self):
                 if isinstance(self.option_definitions[x], Option)]
 
     #--------------------------------------------------------------------------
-    def _create_reference_value_from_links(self, keys, known_keys):
+    def _create_reference_value_options(self, keys, finished_keys):
         """this method steps through the option definitions looking for
         alt paths.  On finding one, it creates the 'reference_value_from' links
         within the option definitions and populates it with copied options."""
         # a set of known reference_value_from_links
-        set_of_reference_value_from_links = set()
-        for key in (k for k in keys if k not in known_keys):
+        set_of_reference_value_option_names = set()
+        for key in keys:
+            if key in finished_keys:
+                continue
             an_option = self.option_definitions[key]
             if an_option.reference_value_from:
 
-                reference_name = '.'.join((
+                fully_qualified_reference_name = '.'.join((
                     an_option.reference_value_from,
                     an_option.name
                 ))
-                if reference_name in self.option_definitions:
+                if fully_qualified_reference_name in keys:
```

Comment:
Why not check against `self.option_definitions` any more?
Seems bigger change than just refactoring.